src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol.cs

```diff
@@ -821,6 +821,22 @@ internal override bool HasSpecialName
             }
         }
 
+        internal override bool HasInternalImplementationOnlyAttribute
+        {
+            get
+            {
+                foreach (var att in this.GetAttributes())
+                {
+                    if (att.IsTargetAttribute("System.Runtime.CompilerServices", "InternalImplementationOnlyAttribute"))
```

Comment:
We should use helper that takes AttributeDescription.InternalImplementationOnlyAttribute so that shapes of constructors are checked and to be consistent with how we check for presence of this attribute in metadata.